packages/code-studio/src/settings/FormattingSectionContent.jsx

```diff
@@ -380,21 +411,44 @@ export class FormattingSectionContent extends PureComponent {
       showTimeZone,
       showTSeparator,
       timeZone,
+      defaultDecimalFormatOptions,
+      defaultIntegerFormatOptions,
     } = this.state;
 
     const formatter = formatSettings
       .filter(FormattingSectionContent.isFormatRuleValidForSave)
       .map(FormattingSectionContent.removeFormatRuleExtraProps);
 
     const { settings, saveSettings } = this.props;
-    saveSettings({
+    const newSettings = {
       ...settings,
       formatter,
       defaultDateTimeFormat,
       showTimeZone,
       showTSeparator,
       timeZone,
-    });
+    };
+    if (
+      defaultDecimalFormatOptions?.defaultFormatString &&
```

Comment:
```suggestion
      defaultDecimalFormatOptions?.defaultFormatString != null &&
```
I think we need to either accept empty string as a valid input here, or mark it as invalid in `render`:
- Open a table with decimals, open settings sidebar
- Change default decimal format to '#abc', confirm the decimal columns get formatted with the new format
- Clear the decimal format input. Now it shows the default placeholder, no red input border, and the columns are still formatted with the previous `#abc` format

Reply:
I think empty should be invalid, just like the other rules.